packages/create-app/templates/default-app/packages/backend/src/plugins/auth.ts

```diff
@@ -18,18 +18,34 @@ export default async function createPlugin(
     providerFactories: {
       ...defaultAuthProviderFactories,
 
-      // This overrides the default GitHub auth provider with a custom one.
-      // Since the options are empty it will behave just like the default
-      // provider, but if you uncomment the `signIn` section you will enable
-      // sign-in via GitHub. This particular configuration uses a resolver
-      // that matches the username to the user entity name. See the auth
-      // documentation for more details on how to enable and customize sign-in:
+      // This replaces the default GitHub auth provider with a customized one.
+      // The `signIn` option enables sign-in for this provider, using the
+      // identity resolution logic that's provided in the `resolver` callback.
+      //
+      // This particular resolver makes all users share a single "guest" identity.
+      // It should only be used for testing and trying out Backstage.
+      //
+      // If you want to use a production ready resolver you can switch to the
+      // the one that is commented out below, it looks up a user entity in the
+      // catalog using the GitHub username of the authenticated user.
+      // That resolver requires you to have user entities populated in the catalog,
+      // for example using https://backstage.io/docs/integrations/github/org
+      //
+      // There other resolvers to choose from, and you can also create
+      // your own, see the auth documentation for more details:
       //
       //   https://backstage.io/docs/auth/identity-resolver
       github: providers.github.create({
-        // signIn: {
-        //   resolver: providers.github.resolvers.usernameMatchingUserEntityName(),
-        // },
+        signIn: {
+          resolver: (_, ctx) =>
+            ctx.issueToken({
```

Comment:
when adding the curly braces I think it'll be educational to also make a `const subject = stringify...` which ensures that the required imports ALSO are helpfully present so you even more easily can get tweakin' without making unnecessary casing mistakes etc. Then the `claims` can be collapsed into one line instead of four perhaps because they'll be so dense already

Reply:
Switched but added in comments for some extra context, so kept it on separate lines